CONTRIBUTING.md

```diff
@@ -0,0 +1,132 @@
+## Contributing
+
+Pull requests for bug fixes are welcome, but before submitting new features or changes to current functionality, [open an issue](https://github.com/DataDog/datadog-ci/issues/new)
+and discuss your ideas or propose the changes you wish to make. After a resolution is reached, a PR can be submitted for review.
+
+### Running command in development environment
+
+When developing the tool, it is possible to run commands using `yarn launch`. It relies on `ts-node`, so does not require building the project for every new change.
+
+```bash
+yarn launch synthetics run-tests --config dev/global.config.json
+```
+
+### Framework and libraries used
+
+- [clipanion](https://github.com/arcanis/clipanion): CLI library to handle the different commands.
+- [eslint](https://github.com/eslint/eslint): Linting ([.eslintrc.js](/.eslintrc.js)).
+- [jest](https://github.com/facebook/jest): Tests are written in Jest.
+- [volta](https://github.com/volta-cli/volta): NodeJS and yarn versioning.
+
+### Repository structure
+
+Commands are stored in the [src/commands](src/commands) folder.
+
+The skeleton of a command is composed of a README, an `index.ts` and a folder for the tests.
+
+```bash
+src/
+└── commands/
+    └── fakeCommand/
+         ├── __tests__/
+         │   └── index.test.ts
+         ├── README.md
+         └── index.ts
+```
+
+Documentation of the command must be placed in the README.md file, the [current README](/README.md) must be updated to link to the new command README.
+
+The `index.ts` file must export classes extending the `Command` class of `clipanion`. The commands of all `src/commands/*/index.ts` files will then be imported and made available in the `datadog-ci` tool.
+
+A sample `index.ts` file for a new command would be:
+
+```typescript
+import {Command} from 'clipanion'
+
+export class HelloWorldCommand extends Command {
+  public async execute() {
+    this.context.stdout.write('Hello world!')
+  }
+}
+
+module.exports = [HelloWorldCommand]
+```
+
+Lastly, test files must be created in the `__tests__/` folder. `jest` is used to run the tests and a CI has been set using GitHub Actions to ensure all tests are passing when merging a Pull Request.
+
+The tests can then be launched through the `yarn test` command, it will find all files with a filename ending in `.test.ts` in the repo and execute them.
+
+### Continuous Integration tests
+
+The CI performs tests to avoid regressions by building the project, running unit tests and running one end-to-end test.
```

Comment:
(nitpick) 
is it just one end-to-end test now? It seems we run e2e for sourcemaps, junit, sarif and sbom too (https://github.com/DataDog/datadog-ci/blob/master/.github/workflows/ci.yml#L86-L112) 

I don't know if we need to dive into details of the synthetic test here: maybe just like to the GHA workflow file and add some comments in there?

Reply:
I agree, this is too much of a deep dive, I'll just remove it as this can be documented internally